lib/Expression_gen.ml

```diff
@@ -0,0 +1,264 @@
+open Core_kernel
+open Mir
+
+let ends_with suffix s =
+  String.is_suffix ~suffix:suffix s
+
+let starts_with prefix s =
+  String.is_prefix ~prefix:prefix s
+
+let functions_requiring_namespace = 
+  [ "e"; "pi"; "log2"; "log10"; "sqrt2"; "not_a_number"; 
+    "positive_infinity"; "negative_infinity"; "machine_precision"; 
+    "abs"; "acos"; "acosh"; "asin"; "asinh"; "atan"; "atan2"; "atanh"; 
+    "cbrt"; "ceil"; "cos"; "cosh"; "erf"; "erfc"; "exp"; "exp2"; "expm1"; 
+    "fabs"; "floor"; "lgamma"; "log"; "log1p"; "log2"; "log10"; "round"; 
+    "sin"; "sinh"; "sqrt"; "tan"; "tanh"; "tgamma"; "trunc"; 
+    "fdim"; "fmax"; "fmin"; "hypot"; "fma" ]
+
+let equality a b = (a = b)
+
+let contains_elt lst elt  = (List.mem lst elt ~equal:equality)
+
+let stan_namespace_qualify f = 
+  if (contains_elt functions_requiring_namespace f) then "stan::math::"^f else f
```

Comment:
Then to check for membership when `functions_requiring_namespace` is a Set, you can use `Set.mem`, as in: `Set.mem functions_requiring_namespace f`